modules/core/src/viewports/globe-viewport.js

```diff
@@ -0,0 +1,156 @@
+import * as vec4 from 'gl-matrix/vec4';
+
+const DEGREES_TO_RADIANS = Math.PI / 180;
+const RADIANS_TO_DEGREES = 180 / Math.PI;
+const EARTH_RADIUS = 6370972;
+const GLOBE_RADIUS = 256;
+
+function getDistanceScales() {
+  const unitsPerMeter = GLOBE_RADIUS / EARTH_RADIUS;
+  const unitsPerDegree = (Math.PI / 180) * GLOBE_RADIUS;
+
+  return {
+    unitsPerMeter: [unitsPerMeter, unitsPerMeter, unitsPerMeter],
+    unitsPerMeter2: [0, 0, 0],
+    metersPerUnit: [1 / unitsPerMeter, 1 / unitsPerMeter, 1 / unitsPerMeter],
+    unitsPerDegree: [unitsPerDegree, unitsPerDegree, unitsPerMeter],
+    unitsPerDegree2: [0, 0, 0],
+    degreesPerUnit: [1 / unitsPerDegree, 1 / unitsPerDegree, 1 / unitsPerMeter]
+  };
+}
+
+export default class GlobeViewport extends Viewport {
+  constructor(opts = {}) {
+    const {
+      latitude = 0,
+      longitude = 0,
+      zoom = 11,
+      nearZMultiplier = 0.1,
+      farZMultiplier = 1
+    } = opts;
+
+    let {width, height, altitude = 1.5} = opts;
+
+    width = width || 1;
+    height = height || 1;
+    altitude = Math.max(0.75, altitude);
+
+    // Calculate view matrix
+    const viewMatrix = new Matrix4().lookAt({eye: [0, -altitude, 0], up: [0, 0, 1]});
+    const scale = Math.pow(2, zoom);
+    viewMatrix.rotateX(latitude * DEGREES_TO_RADIANS);
+    viewMatrix.rotateZ(-longitude * DEGREES_TO_RADIANS);
+    viewMatrix.scale(scale / height);
+
+    const halfFov = Math.atan(0.5 / altitude);
+    const relativeScale = (GLOBE_RADIUS * 2 * scale) / height;
+
+    const viewportOpts = Object.assign({}, opts, {
+      // x, y,
+      width,
+      height,
+
+      // view matrix
+      viewMatrix,
+      longitude,
+      latitude,
+      zoom,
+
+      // projection matrix parameters
+      fovyRadians: halfFov * 2,
+      aspect: width / height,
+      focalDistance: altitude,
+      near: nearZMultiplier,
+      far: Math.min(2, 1 / relativeScale + 1) * altitude * farZMultiplier
+    });
+
+    super(viewportOpts);
+
+    this.distanceScales = getDistanceScales();
+  }
+
+  get projectionMode() {
+    return PROJECTION_MODE.GLOBE;
+  }
+
+  getDistanceScales() {
+    return this.distanceScales;
+  }
+
+  unproject(xyz, {topLeft = true, targetZ} = {}) {
+    const [x, y, z] = xyz;
+
+    const y2 = topLeft ? y : this.height - y;
+    const {pixelUnprojectionMatrix} = this;
+
+    let coord;
+    if (Number.isFinite(z)) {
+      // Has depth component
+      coord = transformVector(pixelUnprojectionMatrix, [x, y2, z, 1]);
+    } else {
+      // since we don't know the correct projected z value for the point,
+      // unproject two points to get a line and then find the point on that line that intersects with the sphere
+      const coord0 = transformVector(pixelUnprojectionMatrix, [x, y2, -1, 1]);
+      const coord1 = transformVector(pixelUnprojectionMatrix, [x, y2, 1, 1]);
+
+      const lt = ((targetZ || 0) / EARTH_RADIUS + 1) * GLOBE_RADIUS;
+      const lSqr = vec3.sqrLen(vec3.sub([], coord0, coord1));
```

Comment:
So projections are no longer linear transformations so we need custom overrides. 

It might be nice to have a WGS84 CARTESIAN project/unproject also built in here in the viewports, which I presume would be more direct operation for this view, rather than going via an ENU system and back.

Reply:
This is more of a question about the `Viewport` API.

Right now we have:

- `projectPosition` world -> common
- `unprojectPosition` common -> world
- `project` world -> screen
- `unproject` screen -> world

Should we add another pair like:

- `projectCommon` common -> screen
- `unprojectCommon` screen -> common